### PR TITLE
Prune install section in .travis.yml; install Dosbox only for make_dos.

### PR DESCRIPTION
**Pull request by arstrube**
This fixes issue #809 - well a related issue which causes all jobs to error when for some reason Dosbox (which is only needed for `make_dos`) cannot be installed.

I tested the github deployer part on my fork and it work (installing correctly, then displaying a message that nothing will be deployed, because it is running on a for - all as expected). After that, I commented out the github deployer part again.

### CODE REVIEW

**Comment by basvodde** [comment]
This causes problems due to the "new hardware" Travis... right?

**Comment by arstrube** [comment]
No, it's not related to the new hardware stuff actually. But it did happen that DosBox could not be installed, and then all jobs would error when only one needed it. So I figured it would be wise to move the "specialty" installs into the build script.

**Comment by arstrube** [comment]
Same goes for Travis Github Deployer.

The update command is issued by the standard apt: block (minuse the --fixmissing, but that doesn't seem to matter anymore)

**Comment by basvodde** [comment]
See: https://docs.travis-ci.com/user/workers/container-based-infrastructure/

**Comment by arstrube** [comment]
In order to not use sudo (with or without my proposed streamlining), we'd have to get them to put DosBox on the whitelist.

Let's say I reduced the sudo-count by 2, if nothing else :-). I think DosBox is the only one left.

**Comment by basvodde** [comment]
Yes, need to get it on the whitelist. Especially as the non-container infra will go away at some point.

**Comment by arstrube** [comment]
At least, while we are still using sudo, only the make_dos job would break, not everything.

File: .travis.yml [hunk 1]
```diff
@@ -34,15 +34,7 @@ global:
 addons:
   apt:
     packages:
-      # - dosbox
```
Comment (basvodde) [inline .travis.yml]: Maybe put this back, as eventually it ought to be here :)

Comment (arstrube) [inline .travis.yml]: Okay, done.

There is a sudo for openwatcom, too, unfortunately, which won't go away unless an apt package were created for openwatcom, or we need to find a way to install the package for the local user. I hear this is possible but somewhat involved.

Comment (basvodde) [inline .travis.yml]: At some point, we'll need to figure it out :) It would speed up the build a lot, I guess.

**Pull request merged**